server/helpers/index.js

```diff
@@ -1,6 +1,189 @@
+            `)
+          }
+        } catch (err) {
+          // Log err
+          reject()
+        }
+        let data = ''
+
+        httpsResponse.on('data', (chunk) => {
+          data += chunk
+        })
+
+        httpsResponse.on('end', () => {
+          try {
+            const response = JSON.parse(data)
+            if (response.success === false) {
+              throw new Error('User not found')
+            }
+
+            // Adds completed kata ids to an array
+            let completed = response.data.map(kata => kata.id)
+            if (response.totalPages === 1) {
+              resolve(completed)
+            } else {
+              // If there is more than one page of results all other pages must
+              // also be requested
+              getHistoricCompletedKatas(userName, response.totalPages)
+                .then((historicCompleted) => {
+                  completed = [...completed, ...historicCompleted]
+                  resolve(completed)
+                })
+                .catch((err) => {
+                  reject(err)
+                })
+            }
+          } catch (err) {
+            // If the user is not found it may indicate that their account has
+            // been deleted
+            // Log err
+            reject()
+          }
+        })
+      }
+    )
+  })
+}
+
+export const updateAllScores = async (katathonId) => {
+  try {
+    const katathon = await Katathon.findById(katathonId)
+    const { users, katas } = katathon
+    if (!users.length || !katas.length) {
+      return
+    }
+    // Compiles a list of all ongoing userScore requests. Once they have all
+    // updated and resolved the new data is written to db
+    const allPromises = []
+    users.forEach((user) => {
+      // This promise must always resolve so that if one user request fails the
+      // other scores will still be updated. Currently a deleted account will fail
+      const promise = new Promise(async (resolve) => {
+        try {
+          const completed = await getCompletedKatas(user.userName)
+
+          // This checks that no requests for completed katas have failed
+          if (!Array.isArray(completed)) {
+            throw new Error(`Failed to get completed katas for ${user.userName}`)
+          }
+
+          // Loops through katas and adds score when they are found in completed
+          let userScore = 0
+          katas.forEach((kata) => {
+            if(completed.includes(kata.kataId)) {
+              userScore += kata.score
+            }
+          })
+
+          // Updates model where appropriate
+          if (userScore !== user.userScore) {
+            katathon.users = katathon.users.map((existingUser) => {
+              return existingUser.userName === user.userName
+                ? { ...existingUser._doc, userScore }
+                : existingUser
+            })
+          }
+          resolve()
+        } catch (err) {
+          // Log err
+          resolve()
```

Comment:
Why call resolve() in the catch? Shouldn't we reject if an error is caught?

Reply:
Has to be a resolve as it is inside a Promise.all. If one user fails (because they delete their account for instance) then we still want to get the other users' scores